php/Blocks/TemplateEditor.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * TemplateEditor.
+ *
+ * @package   Genesis\CustomBlocks
+ * @copyright Copyright(c) 2021, Genesis Custom Blocks
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Genesis\CustomBlocks\Blocks;
+
+/**
+ * Class TemplateEditor
+ */
+class TemplateEditor {
+
+	/**
+	 * The block names that have had their CSS rendered.
+	 *
+	 * @var string[]
+	 */
+	public $blocks_with_rendered_css = [];
+
+	/**
+	 * Renders markup that was entered in the template editor.
+	 *
+	 * @param string $markup The markup to render.
+	 */
+	public function render_markup( $markup ) {
+		$rendered = preg_replace_callback(
+			'#{{(\S+?)}}#',
+			static function( $matches ) {
+				ob_start();
+				block_field( $matches[1] );
+				return ob_get_clean();
+			},
+			$markup
+		);
+
+		// Escape characters before { should be stripped, like \{\{example\}\}.
+		// Like if they have a tutorial on Mustache and need the template to render {{example}}.
+		echo preg_replace( '#\\\{\\\{(\S+?)\\\}\\\}#', '{{\1}}', $rendered ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
That's a pretty simple regex pattern, nice.

Reply:
I do wonder if this could use escaping as well. Just concerned that security researchers might have an issue with this, or even the core plugins team.

Reply:
How about https://github.com/studiopress/genesis-custom-blocks/pull/69/commits/df28138b0c5e5bee574f66c3205cb81dca4c8bfc? It's escaped with `wp_kses_post()`.